packages/datagateway-common/src/table/columnFilters/dateColumnFilter.component.tsx

```diff
@@ -224,45 +249,33 @@ const DateColumnFilter = (props: DateColumnFilterProps): React.ReactElement => {
                 onChange: props.onChange,
               });
             }}
-            componentsProps={{
+            onError={(newError) => setError(newError)}
+            slots={{
+              actionBar: CustomClearButton,
+              textField: CustomTextField,
+            }}
+            slotProps={{
               actionBar: {
                 actions: ['clear'],
               },
-            }}
-            components={{
-              ActionBar: CustomClearButton,
-            }}
-            OpenPickerButtonProps={{
-              size: 'small',
-              'aria-label': `${props.label} filter from, date picker`,
-            }}
-            // id={props.label + 'filter from'}
-            renderInput={(renderProps) => {
-              const error =
-                // eslint-disable-next-line react/prop-types
-                (renderProps.error || invalidDateRange) ?? undefined;
-              let helperText = 'Date format: yyyy-MM-dd.';
-              if (invalidDateRange) helperText = 'Invalid date range';
-
-              return (
-                <TextField
-                  {...renderProps}
-                  id={props.label + ' filter from'}
-                  inputProps={{
-                    ...renderProps.inputProps,
-                    placeholder: 'From...',
-                    'aria-label': `${props.label} filter from`,
-                  }}
-                  variant="standard"
-                  error={error}
-                  {...(error && { helperText: helperText })}
-                />
-              );
+              textField: {
+                inputProps: {
+                  invalidDateRange,
+                  errorText,
+                  filterByTime: props.filterByTime,
+                  id: props.label + ' filter from',
+                  placeholder: 'From...',
+                  'aria-label': `${props.label} filter from`,
+                },
+              },
+              openPickerButton: {
+                size: 'small',
+                'aria-label': `${props.label} filter from, date picker`,
+              },
             }}
           />
           <DatePicker
```

Comment:
We should probably explicitly set `view={["year","month","day"]}` as othewise a user can't select the month other than by pressing the arrows. This needs to be done for all `DatePickers` (`DateTimePickers` are fine since we set the views there anyway...)